phpunit/imap/MailCollectorTest.php

```diff
@@ -1282,4 +1282,50 @@ public function testGetAdditionnalMailServerProtocols(
             $this->assertNull($storage);
         }
     }
+
+
+
+    public static function cleanContentProvider()
+    {
+        return [
+            [
+                'filter' => '<a href="mailto:test@mail.local%3cmailto:test@mail.local">test@mail.local&lt;mailto:test@mail.local&gt;</a>',
+                'mail_content' => 'Text before<a href="mailto:test@mail.local%3cmailto:test@mail.local">test@mail.local&lt;mailto:test@mail.local&gt;</a>Text after',
+                'filtered' => true,
+            ],
+            [
+                'filter' => '<a href="mailto:test@mail.local%3cmailto:test@mail.local">test@mail.local&lt;mailto:test@mail.local</a>&gt;',
+                'mail_content' => 'Text before<a href="mailto:test2@mail.local%3cmailto:test2@mail.local">test2@mail.local&lt;mailto:test2@mail.local&gt;</a>Text after',
+                'filtered' => false,
+            ],
+        ];
+    }
+
+
+    /**
+     * @dataProvider cleanContentProvider
+     */
+    public function testCleanContent(
+        string $filter,
+        string $mail_content,
+        bool $filtered
+    ) {
+        $this->createItem(
+            'BlacklistedMailContent',
+            [
+                'name' => __METHOD__,
+                'content' => $filter
+            ]
+        );
+
+        $original = $mail_content;
+
+        $mailcollector = new \MailCollector();
+        $result = $mailcollector->cleanContent($original);
+        if ($filtered) {
+            $this->assertNotEquals($result, $original);
+        } else {
+            $this->assertEquals($result, $original);
+        }
```

Comment:
Testing a not equals does not seems the best idea here. I guess you should just test the expected string is present in both cases.

Seems correct appart of that